api-tests/val/nspe/val_dispatcher.c

```diff
@@ -68,37 +69,37 @@ __attribute__((unused)) static void val_print_api_version(void)
 **/
 val_status_t val_test_load(test_id_t *test_id, test_id_t test_id_prev)
 {
-    int             i;
     val_test_info_t test_list[] = {
 #include "test_entry_list.inc"
                                   {VAL_INVALID_TEST_ID, NULL}
                                   };
+    val_test_info_t *test_info = &test_list[0];
 
-    for (i = 0; i < (int)(sizeof(test_list)/sizeof(test_list[0])); i++)
+    if (test_id_prev != VAL_INVALID_TEST_ID)
     {
-        if (test_id_prev == VAL_INVALID_TEST_ID)
+        for(; test_info->test_id != VAL_INVALID_TEST_ID; test_info++)
         {
-            *test_id = test_list[i].test_id;
-            g_test_info_addr = (addr_t) test_list[i].entry_addr;
-            return VAL_STATUS_SUCCESS;
-        }
-        else if (test_id_prev == test_list[i].test_id)
-        {
-            *test_id = test_list[i+1].test_id;
-            g_test_info_addr = (addr_t) test_list[i+1].entry_addr;
-            return VAL_STATUS_SUCCESS;
+            if (test_info->test_id == test_id_prev)
+            {
+                test_info++;
+                break;
+            }
         }
-        else if (test_list[i].test_id == VAL_INVALID_TEST_ID)
+    }
+
+    for(; test_info->test_id != VAL_INVALID_TEST_ID; test_info++)
+    {
+        if (pal_is_test_enabled(VAL_GET_TEST_NUM(test_info->test_id)))
```

Comment:
I think removing VAL_GET_TEST_NUM conversion would be more intuitive. In the logs the full test number (e.g 201, 202, etc) is used so users will not know that the number has to be converted without checking the sources.

Reply:
I don't know anything about the future changes of val.h, but from our point of view it does not really matter. At the end there are tests that have unique ID-s that can be extracted from the logs so if someone wants to disable them will use these values. In our case these filters will be just temporary until the disabled tests are fixed so it does not matter if the bases change in the future.
Thank you for the commit, tomorrow I will test it, but seems to be exactly what I was looking for. :)

Reply:
Tested the update and it works as expected. Thank you!